Software/grab/include/GrabberBase.hpp

```diff
@@ -40,26 +44,33 @@ enum GrabResult {
 };
 
 struct ScreenInfo {
-    ScreenInfo()
-        : handle(NULL)
-    {}
-    QRect rect;
-    void * handle;
-    bool operator== (const ScreenInfo &other) const {
+    ScreenInfo() = default;
+
+    bool operator==(const ScreenInfo &other) const {
         return other.rect == this->rect;
     }
+
+    QRect rect;
+    void * handle = nullptr;
 };
 
 struct GrabbedScreen {
-    GrabbedScreen()
-        : imgFormat(BufferFormatUnknown)
-        , associatedData(NULL)
-    {}
-    unsigned char * imgData;
-    size_t imgDataSize;
-    BufferFormat imgFormat;
+    GrabbedScreen() = default;
+
+#ifdef MAC_OS_CG_GRAB_SUPPORT
+    CGImageRef displayImageRef = nullptr;
+    CFDataRef imageDataRef = nullptr;
+
+    // This is only a view to the |imageDataRef| pixels raw data.
+    const unsigned char * imgData = nullptr;
+#else
+    unsigned char * imgData = nullptr;
+#endif
```

Comment:
The preferred (or rather designed) method of using this is the `associatedData` pointer to be used by the grabber. That is, create a custom struct/object in your grabber and point to it from associatedData. See `DDuplScreenData` for reference